create_shards.c

```diff
@@ -73,6 +80,11 @@ master_create_distributed_table(PG_FUNCTION_ARGS)
 	AttrNumber partitionColumnId = InvalidAttrNumber;
 	char *partitionColumnName = text_to_cstring(partitionColumnText);
 	char *tableName = text_to_cstring(tableNameText);
+	Var* partitionColumnVar = NULL;
+	Oid partitionColumnTypeId = InvalidOid;
+	Oid partitionColumnOpClassId = InvalidOid;
```

Comment:
A `partitionColumnOpClassId` variable is no longer necessary if we move the `ereport` call into `GetSupportRoutine`, so we should kill it entirely. Additionally, the `btreeSupportRoutine` and `hashSupportRoutine` variables should have their scope reduced by moving their declaration into the relevant blocks below.